PersonApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -47,16 +40,17 @@ public async Task<Person> PostNewPersonAsync(CreatePersonRequestObject requestOb
         }
 
         [LogCall]
-        public async Task<Person> UpdatePersonByIdAsync(UpdatePersonRequestObject requestObject)
+        public async Task<Person> UpdatePersonByIdAsync(UpdatePersonRequestObject requestObject, PersonQueryObject query)
         {
-            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync to update id {requestObject.Id}");
+            _logger.LogDebug($"Calling IDynamoDBContext.SaveAsync to update id {query.Id}");
 
-            var load = await _dynamoDbContext.LoadAsync<PersonDbEntity>(requestObject.Id).ConfigureAwait(false);
+            var personDbEntity = requestObject.ToDatabase();
+            personDbEntity.Id = query.Id;
+            var load = await _dynamoDbContext.LoadAsync<PersonDbEntity>(query.Id).ConfigureAwait(false);
 
             if (load == null) return null;
 
-            var personDbEntity = requestObject.ToDatabase();
-            await _dynamoDbContext.SaveAsync(personDbEntity).ConfigureAwait(false);
+            await _dynamoDbContext.SaveAsync(personDbEntity, new DynamoDBOperationConfig { IgnoreNullValues = true }).ConfigureAwait(false);
```

Comment:
Why do we need to ignore null values here?
Is it something to do with potentially having null sub-object properties?

Reply:
IgnoreNullValues is required to prevent the API from updating values that don't need to actually be updated, hence the frontend doesn't need to provide the full object in the request